### PR TITLE
Add typing annotations for torch.nn.quantized.dynamic.modules.rnn

### LINKED ISSUE #43185: Enable torch.nn.quantized.dynamic.modules.rnn typechecks during CI
`TestTypeHints.test_run_mypy_strict` should pass with `[mypy-torch.nn.quantized.dynamic.modules.rnn]` deleted from mypy.ini

cc @ezyang @malfet @rgommers

### PR DESCRIPTION
**Pull request by guilhermeleobas**
Fixes #43185

xref: [gh-43072](https://github.com/pytorch/pytorch/issues/43072)

### CODE REVIEW

**Comment by hameerabbasi** [comment]
Probably rebase or merge `upstream/viable`.

File: torch/nn/quantized/dynamic/modules/rnn.py [hunk 1]
```diff
@@ -661,13 +668,12 @@ class LSTMCell(RNNCellBase):
     """
 
     def __init__(self, *args, **kwargs):
-        super(LSTMCell, self).__init__(*args, num_chunks=4, **kwargs)
+        super(LSTMCell, self).__init__(*args, **kwargs)
```
[the remark this answers is not in the document]
Comment (guilhermeleobas) [reply torch/nn/quantized/dynamic/modules/rnn.py:671]: Done!

**Comment by hameerabbasi** [comment]
Perhaps merge `viable/strict`?

**Comment by guilhermeleobas** [comment]
> It seems that the failures were introduced in this PR. I can't reproduce because it requires `fbgemm`:
> 
> Command to reproduce test failure:
> 
> ```shell
> $ pytest test/test_quantization.py -k 'test_quantized_rnn' -sv -rs
> 
> platform linux -- Python 3.8.4, pytest-5.4.3, py-1.9.0, pluggy-0.13.1 -- /home/guilhermel/miniconda3/envs/pytorch-cuda-dev/bin/python
> cachedir: .pytest_cache
> hypothesis profile 'default' -> database=DirectoryBasedExampleDatabase('/work/guilhermel/git/Quansight/pytorch/.hypothesis/examples')
> rootdir: /work/guilhermel/git/Quansight/pytorch
> plugins: hypothesis-5.20.3
> collected 323 items / 321 deselected / 2 selected
> 
> test/test_quantization.py::TestPostTrainingDynamic::test_quantized_rnn SKIPPED
> test/test_quantization.py::TestPostTrainingDynamic::test_quantized_rnn_cell SKIPPED
> 
> ======================================================================================================== short test summary info ========================================================================================================
> SKIPPED [1] test/quantization/test_quantize.py:737: Quantized operations require FBGEMM. FBGEMM is only optimized for CPUs with instruction set support AVX2 or newer.
> SKIPPED [1] test/quantization/test_quantize.py:778: Quantized operations require FBGEMM. FBGEMM is only optimized for CPUs with instruction set support AVX2 or newer.
> ```

I don't think this test are related to the changes introduced in this pull request. I can reproduce the same error on `master` and `viable/strict`.

**Comment by rgommers** [comment]
> I don't think this test are related to the changes introduced in this pull request. I can reproduce the same error on `master` and `viable/strict`.

You mean in CI on this PR? https://ezyang.github.io/pytorch-ci-hud/build/pytorch-master looks very green right now. It's likely caused by this PR.

The 11 JIT failures in `pytorch_linux_bionic_py3_6_clang9_test` look unrelated, but would be nice to be sure.

**Comment by rgommers** [comment]
> You mean in CI on this PR? https://ezyang.github.io/pytorch-ci-hud/build/pytorch-master looks very green right now. It's likely caused by this PR.

I rebuilt with fbgemm support and tested this branch - the two hypothesis tests pass for me locally.

**Comment by rgommers** [comment]
@guilhermeleobas other PRs don't have these failures, so some more digging is needed. For starters, can you squash all your commits and rebase on current master?

**Comment by guilhermeleobas** [comment]
The error was related to pytorch not parsing `# type: ignore` comments correctly. PR #43446 seems to fix that. I will rebase and run CI again.

**Review by rgommers — APPROVED** [review]
LGTM now, thanks @guilhermeleobas

**Comment by rgommers** [comment]
The two CI failures are unrelated.

**Comment by guilhermeleobas** [comment]
Thanks for the review, @rgommers and @hameerabbasi

**Review by malfet — APPROVED** [review]

**Pull request closed without merging**